examples/lines_bars_and_markers/angle_marker.py

```diff
@@ -0,0 +1,323 @@
+            self.text.set_position([offs*np.cos(angle), offs*np.sin(angle)])
+
+
+#########################################################################
+# .. _angle-marker-usage:
+#
+# Usage
+# ~~~~~
+#
+# Required arguments to ``AngleMarker`` are the center of the arc, ``xy``,
+# and two points, such that the arc spans between the two vectors
+# connecting p1 and p2 with xy, respectively. Those are given in data
+# coordinates.
+# Further arguments are the ``size`` of the arc and its ``unit``.
+# Additionally, a ``text`` can be specified, that will be drawn either in- or
+# outside of the arc, according to the value of ``textposition``.
+# Usage of those arguments is shown below.
+
+fig, (ax1, ax2, ax3) = plt.subplots(nrows=3, sharex=True, figsize=(7, 7),
+                                    gridspec_kw=dict(height_ratios=(3, 1, 1)))
+ax2.margins(y=0.4)
+ax3.margins(y=0.4)
+fig.canvas.draw()  # Need to draw the figure to define renderer
+
+#### SUBPLOT 1 ####
+# Plot two crossing lines and label each angle between them with the
+# above `AngleMarker`tool.
+center = (4.5, 650)
+p1 = [(2.5, 710), (6.0, 605)]
+p2 = [(3.0, 275), (5.5, 900)]
+line1, = ax1.plot(*zip(*p1))
+line2, = ax1.plot(*zip(*p2))
+point, = ax1.plot(*center, marker="o")
+
+am1 = AngleMarker(center, p1[1], p2[1], ax=ax1, size=75, text=r"$\alpha$")
+am2 = AngleMarker(center, p2[1], p1[0], ax=ax1, size=35, text=r"$\beta$")
+am3 = AngleMarker(center, p1[0], p2[0], ax=ax1, size=75, text=r"$\gamma$")
+am4 = AngleMarker(center, p2[0], p1[1], ax=ax1, size=35, text=r"$\theta$")
+
+
+# Showcase some styling options for the angle arc, as well as the text
+p = [(6.0, 400), (5.3, 410), (5.6, 300)]
+ax1.plot(*zip(*p))
+am5 = AngleMarker(p[1], p[0], p[2], ax=ax1, size=40, text=r"$\Phi$",
+                  linestyle="--", color="gray", textposition="outside",
+                  text_kw=dict(fontsize=16, color="gray"))
+
+
+#### SUBPLOT 2 ####
+# Helper function to draw angle easily
+def plot_angle(ax, pos, angle, length=0.95, acol="C0", **kwargs):
+    vec2 = np.array([np.cos(np.deg2rad(angle)), np.sin(np.deg2rad(angle))])
+    xy = np.c_[[length, 0], [0, 0], vec2*length].T + np.array(pos)
+    ax.plot(*xy.T, color=acol)
+    return AngleMarker(pos, xy[0], xy[2], ax=ax, **kwargs)
+
+# Showcase different textpositions
+kw = dict(size=75, unit="points", text=r"$60°$")
+
+am6 = plot_angle(ax2, (2.0, 0), 60, textposition="inside", **kw)
+am7 = plot_angle(ax2, (3.5, 0), 60, textposition="outside", **kw)
+am8 = plot_angle(ax2, (5.0, 0), 60, textposition="edge",
+                 text_kw=dict(bbox=dict(boxstyle="round", fc="w")), **kw)
+am9 = plot_angle(ax2, (6.5, 0), 60, textposition="edge",
+                 text_kw=dict(xytext=(30, 20), arrowprops=dict(arrowstyle="->",
+                              connectionstyle="arc3,rad=-0.2")), **kw)
+
+ax2.annotate("textpostion", xy=(.02, .9), xycoords="axes fraction",
+                 bbox=dict(boxstyle="round", fc="w"), ha="left")
```

Comment:
Would go for

~~~
ax2.annotate("textpostion", xy=(.02, 1), xycoords="axes fraction",
             bbox=dict(boxstyle="round", fc="w"), ha="left", va="center")
~~~
to not have a "magic" .9 there and to make it work independent of text size.